File: go/nth-prime/nth_prime.go
package prime

import "math"

var primes = []int{2}

// Nth returns the nth prime.
func Nth(nth int) (int, bool) {
	count := len(primes)

	for num := primes[count-1] + 1; nth > 0; num++ {
		if count >= nth {
			return primes[nth-1], true
		}
		if isPrime(num) {
			primes = append(primes, num)
			count++
		}
	}

	return 0, false
}

// isPrime checks is a given number is prime.
func isPrime(number int) bool {
	limit := int(math.Sqrt(float64(number)))

	for n := 2; n <= limit; n++ {
		if number%n == 0 {
			return false
		}
	}

	return true
}

File: javascript/nth-prime/nth-prime.js
export class Prime {
    *primes() {
        yield 2;
        for (let p = 3; true; p += 2)
            if (this.isPrime(p))
                yield p;
    }

    isPrime(p) {
        if (p % 2 === 0)
            return false;

        const max = Math.sqrt(p);
        for (let n = 3; n <= max; n += 2)
            if (p % n === 0)
                return false;

        return true;
    }

    nth(n) {
        if (n === 0)
            throw new Error('Prime is not possible');

        let count = 0;
        for (const prime of this.primes())
            if (++count === n)
                return prime;
    }
}
